streams/src/test/java/org/apache/kafka/streams/integration/KStreamRepartitionIntegrationTest.java

```diff
@@ -0,0 +1,859 @@
+        final String repartitionName = "new-topic";
+        final long timestamp = System.currentTimeMillis();
+
+        sendEvents(
+            timestamp,
+            Arrays.asList(
+                new KeyValue<>(1, "A"),
+                new KeyValue<>(2, "B")
+            )
+        );
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        builder.stream(inputTopic, Consumed.with(Serdes.Integer(), Serdes.String()))
+               .repartition(Repartitioned.as(repartitionName))
+               .groupByKey()
+               .count()
+               .toStream()
+               .to(outputTopic);
+
+        startStreams(builder);
+
+        validateReceivedMessages(
+            new IntegerDeserializer(),
+            new LongDeserializer(),
+            Arrays.asList(
+                new KeyValue<>(1, 1L),
+                new KeyValue<>(2, 1L)
+            )
+        );
+
+        final String repartitionTopicName = toRepartitionTopicName(repartitionName);
+
+        assertTrue(topicExists(repartitionTopicName));
+        assertEquals(4, getNumberOfPartitionsForTopic(repartitionTopicName));
+    }
+
+    @Test
+    public void shouldCreateOnlyOneRepartitionTopicWhenRepartitionIsFollowedByGroupByKey() throws ExecutionException, InterruptedException {
+        final String repartitionName = "new-partitions";
+        final long timestamp = System.currentTimeMillis();
+
+        sendEvents(
+            timestamp,
+            Arrays.asList(
+                new KeyValue<>(1, "A"),
+                new KeyValue<>(2, "B")
+            )
+        );
+
+        final StreamsBuilder builder = new StreamsBuilder();
+
+        final Repartitioned<String, String> repartitioned = Repartitioned.<String, String>as(repartitionName)
+            .withKeySerde(Serdes.String())
+            .withValueSerde(Serdes.String())
+            .withNumberOfPartitions(1);
+
+        builder.stream(inputTopic, Consumed.with(Serdes.Integer(), Serdes.String()))
+               .selectKey((key, value) -> key.toString())
+               .repartition(repartitioned)
+               .groupByKey()
+               .count()
+               .toStream()
+               .to(outputTopic);
+
+        startStreams(builder);
+
+        final String topology = builder.build().describe().toString();
+
+        validateReceivedMessages(
+            new StringDeserializer(),
+            new LongDeserializer(),
+            Arrays.asList(
+                new KeyValue<>("1", 1L),
+                new KeyValue<>("2", 1L)
+            )
+        );
+
+        assertTrue(topicExists(toRepartitionTopicName(repartitionName)));
+        assertEquals(1, countOccurrencesInTopology(topology, "Sink: .*-repartition"));
+    }
+
+    @Test
+    public void shouldGenerateRepartitionTopicWhenNameIsNotSpecified() throws ExecutionException, InterruptedException {
```

Comment:
Seems to be unit-test able via `Topology#describe()` ?

Reply:
Thought about that, but somehow it felt "safer" with integration tests. Mainly because I was more comfortable verifying that topics actually get created when using repartition operation.